openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -107,7 +107,9 @@ def test_get_programs_for_dashboard(self):
             for course_code in program['course_codes']:
                 for run in course_code['run_modes']:
                     course_key = run['course_key']
-                    expected[course_key] = program
+                    if not expected.get(course_key, None):
+                        expected[course_key] = []
+                    expected[course_key].append(program)
```

Comment:
`setdefault` can be used here as well (see https://github.com/edx/edx-platform/pull/11725/files#r54675360)